src/librustc_resolve/lib.rs

```diff
@@ -220,9 +220,10 @@ fn resolve_struct_error<'sess, 'a>(resolver: &'sess Resolver,
             let sugg_msg = "try using a local type parameter instead";
             if let Some((sugg_span, new_snippet)) = cm.generate_local_type_param_snippet(span) {
                 // Suggest the modification to the user
-                err.span_suggestion(sugg_span,
+                err.span_suggestion_with_applicability(sugg_span,
                                     sugg_msg,
-                                    new_snippet);
+                                    new_snippet,
+                                    Applicability::MachineApplicable);
```

Comment:
Can you fix the formatting as you did below? (each arg on it's own line, 4 spaces indented, trailing comma on every line, closing `);` on it's own line)